internal/cmd/io_binary.go

```diff
@@ -0,0 +1,242 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/pkg/errors"
+)
+
+const (
+	binaryPipeFmt        = `\\.\pipe\binary-%s-%s`
+	binaryCmdWaitTimeout = 5 * time.Second
+)
+
+func newBinaryCmd(ctx context.Context, uri *url.URL, envs []string) *exec.Cmd {
+	if uri.Host == "" && uri.Path == "" {
+		return nil
```

Comment:
Shouldn't we treat this as an error condition? The user asked for binary logging but didn't provide a path to a binary. Might be missing something though